xamarin/Xamarin.Forms
# [Android] Numeric keyboard should be able to process all available keys

### Description of Change

Android numeric keyboard displayed 14 keys (https://developer.xamarin.com/recipes/cross-platform/xamarin-forms/controls/choose-keyboard-for-entry/), but only the digits along with the dot character were processed. This ensures that all characters can be input.

Steps to verify changes:
- Install Google Keyboard from the play store
- Change your input language to Swedish
- Run sample code
- Initially, `KeyListener` should listen for the 14 keys.
- Change keyboard type by tapping on the button
- `KeyListener` should be set to the proper listener

`InputType = InputTypes.ClassNumber | InputTypes.NumberFlagSigned | InputTypes.NumberFlagDecimal` does not work (tested on S6 / M) probably because of Android fragmentation. 

IMO, entry validation should be left to the developer (see user's request). The simplest way to validate numbers is with `double.Parse(text, CultureInfo.CurrentCulture)`, but for whatever reason custom validation might be required or the UI might be tasked to react as the entry changes from a valid number to an invalid number or vice versa.

You might see [this bug](https://bugzilla.xamarin.com/show_bug.cgi?id=43867) while testing the sample. Please note that this is an existing bug.
### Bugs Fixed
- https://bugzilla.xamarin.com/show_bug.cgi?id=42000
